From 09e2349c51cd9a8e5d7544e07266a5ff681ef328 Mon Sep 17 00:00:00 2001
From: Norbert Thiebaud <nthiebaud@gmail.com>
Date: Fri, 15 Nov 2013 16:43:20 +0000
Subject: Save downloaded tarballs under external/tarballs/

The choice of src/ as a place to save those tarballs was unfortunate,
since it is used everywhere else for own source code. After moving all
external modules to external/, it makes sense to put the tarballs there
too and, while at it, use a more appropriate name.

If src exist as a directory, we move it to external/tarballs and leave
a link behind for backward compat for people switching from after to
before the change in the git repo.

Change-Id: I045c4470945f5fadf7760855833ef58e6594ec80
Reviewed-on: https://gerrit.libreoffice.org/6597
Reviewed-by: Björn Michaelsen <bjoern.michaelsen@canonical.com>
Tested-by: Björn Michaelsen <bjoern.michaelsen@canonical.com>
---
diff --git a/.gitignore b/.gitignore
index 0de9054..a227af3 100644
--- a/.gitignore
+++ b/.gitignore
@@ -16,7 +16,7 @@
 # if that is not the case, don't hide it here, but try to fix it
 #
 # The intent is that after a make distclean, .gitignore
-# should not catch anything but possibly /src
+# should not catch anything but possibly /external/tarballs
 #
 
 # where stuff are built
@@ -24,7 +24,7 @@
 /workdir
 
 # where the external downloads maybe located
-/src
+/external/tarballs
 
 # autoconf generated stuff
 /aclocal.m4
diff --git a/configure.ac b/configure.ac
index fd47321..dc7be27 100644
--- a/configure.ac
+++ b/configure.ac
@@ -4669,7 +4669,11 @@ AC_SUBST([SYSTEM_FONTCONFIG])
 dnl whether to find & fetch external tarballs?
 dnl ===================================================================
 if test -z "$TARFILE_LOCATION"; then
-    TARFILE_LOCATION="$SRC_ROOT/src"
+    if test -d "$SRC_ROOT/src" ; then
+        mv "$SRC_ROOT/src" "$SRC_ROOT/external/tarballs"
+        ln -s "$SRC_ROOT/external/tarballs" "$SRC_ROOT/src"
+    fi
+    TARFILE_LOCATION="$SRC_ROOT/external/tarballs"
 fi
 AC_SUBST(TARFILE_LOCATION)
 
--
cgit v0.9.0.2-2-gbebe
